### PR TITLE
Switch by default to new flavour tool based on Hadrons. Update of exampl...

### PR DESCRIPTION
**Pull request by acaudron**
...e scripts.

Switch for the Btag DQM to the new jet flavour tool described here: 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideBTagMCTools#Hadron_based_jet_flavour_definit
and already default in PAT.
Examples scripts are also updated accordingly.
Possibility to use the old flavour tool is conserved, especially as the new tool, targeting PFJets, cannot be applied on some jet collections as CaloJets.

### CODE REVIEW

**Comment by deguio** [comment]
+1

File: Validation/RecoB/plugins/BTagPerformanceAnalyzerMC.cc [hunk 1]
```diff
@@ -243,17 +245,37 @@ void BTagPerformanceAnalyzerMC::analyze(const edm::Event& iEvent, const edm::Eve
...
+    for (JetFlavourInfoMatchingCollection::const_iterator iter = jetMC->begin(); 
+	 iter != jetMC->end(); ++iter) {
+      unsigned int fl = std::abs(iter->second.getPartonFlavour());
+      flavours.insert(std::make_pair(iter->first, fl));
+      const GenParticleRefVector &lep = iter->second.getLeptons();
+      reco::JetFlavour::Leptons lepCount;
+      for (unsigned int i=0; i<lep.size(); i++){
+	if(abs(lep.at(0)->pdgId())==11) lepCount.electron++;
+	else if(abs(lep.at(0)->pdgId())==13) lepCount.muon++;
+	else if(abs(lep.at(0)->pdgId())==15) lepCount.tau++;
```
Comment (davidlange6) [inline Validation/RecoB/plugins/BTagPerformanceAnalyzerMC.cc]: please change all the .at(0) to [0]

**Comment by acaudron** [comment]
I updated the branch. Also it allowed me to see I made a mistake: the index should have been i and not 0.

**Comment by deguio** [comment]
+1

File: DQMOffline/RecoB/python/bTagSequences_cff.py [hunk 2]
```diff
@@ -20,11 +18,20 @@
 softPFElectronsTagInfos.jets = jetID
 
 #for MC do the matching with you jet collection
+from PhysicsTools.JetMCAlgos.HadronAndPartonSelector_cfi import selectedHadronsAndPartons
+from PhysicsTools.JetMCAlgos.AK4PFJetsMCFlavourInfos_cfi import ak4JetFlavourInfos
+ak4JetFlavourInfos.jets = jetID
+
```
Comment (davidlange6) [inline DQMOffline/RecoB/python/bTagSequences_cff.py]: please make a clone of the ak4JetFlavourInfos module before changing it

**Comment by acaudron** [comment]
DQMOffline/RecoB/python/bTagSequences_cff.py was only used in our examples cfg: it was made when ak5PFJetsCHS was not default for b-tagging in RECO and when JEC was not implemented in the b-tagging DQM analysers. So it was a way to have a lighter cfg's.
Now it was not really useful anymore so I removed it and update the examples accordingly.

This update has no impact on sequences ran centrally.

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**